nft_ingester/src/transaction_notifications.rs

```diff
@@ -84,21 +83,29 @@ async fn handle_transaction(manager: Arc<ProgramTransformer>, item: RecvData) ->
             statsd_count!("ingester.seen", 1, "stream" => TRANSACTION_STREAM);
         }
         let seen_at = Utc::now();
-        statsd_time!(
-            "ingester.bus_ingest_time",
-            (seen_at.timestamp_millis() - tx.seen_at()) as u64,
-            "stream" => TRANSACTION_STREAM
-        );
+        metric! {
+            statsd_time!(
+                "ingester.bus_ingest_time",
+                (seen_at.timestamp_millis() - tx.seen_at()) as u64,
+                "stream" => TRANSACTION_STREAM
+            );
+        }
+
         let begin = Instant::now();
         let res = manager.handle_transaction(&tx).await;
-        ret_id = capture_result(
-            id,
+        let should_ack = capture_result(
```

Comment:
why do we need the should_ack, for clarity?

Reply:
Yep, simply for clarity. At one point we were fiddling with some metrics on our end and ended up breaking this code without realizing it. The refactor makes it much more explicit.